virtinst/guest.py

```diff
@@ -845,6 +845,37 @@ def _convert_to_vnc_graphics(self):
             dev.add_child(listen)
         dev.set_defaults(self)
 
+    def _convert_to_vnc_video(self):
+        """
+        If there's no video device, add a default one.
+        If there's any qxl device, we rip out every video device and
+            re-add a default one in place
+        We unconditionally disable accel3d if it's enabled.
+        """
+        if not self.devices.video:
+            videodev = DeviceVideo(self.conn)
+            videodev.set_defaults(self)
+            self.add_device(videodev)
+            return
+
+        for video in self.devices.video:
+            # Disable accel3d which is largely tied to spice
+            video.accel3d = None
```

Comment:
Not sure about this. For VNC we can add `egl-headless` graphics type which makes virgl work with VNC.

Reply:
Good call. I did not understand `egl-headless` until 30 mins ago :) Will fix